### PR TITLE
Enable wasm workers to run under node

### PR DESCRIPTION
**Pull request by sbc100**
This means we can run the tests without depending on the browser.

### CODE REVIEW

File: src/library_wasm_worker.js
Comment (kripken) [inline src/library_wasm_worker.js:142]: `addEventListener` doesn't work under node?

**Review by kripken — APPROVED** [review]

File: src/library_wasm_worker.js
Comment (sbc100) [reply src/library_wasm_worker.js:142]: Our polyfill in `src/shell.js` does not support this no.  We could add to the polyfill, but probably better to be consistent in how we attach the onmessage handler (this is how we do it in the pthread code)

**Comment by sbc100** [comment]
Will wait for @juj for final approval

**Comment by sbc100** [comment]
Actually I'm going to land this now since it seems fairly straight forward, and I could use this for debugging some other PRs.  If the `onmessage` part of the change is controversial at all we can address that in a followup.

**Comment by kleisauke** [comment]
Interesting, I remembered that more changes were needed to fix Node.js compatibility, see e.g. PR #18201 (I should have split that PR back then, sorry).

**Comment by sbc100** [comment]
This change is minimal.. just enough to get the first few tests passing.  There will more changes needed I'm sure to get the rest of them running.

File: src/library_wasm_worker.js
Comment (juj) [reply src/library_wasm_worker.js:142]: Using addEventListener here was very deliberate, because addEventListener can be used many times, but there can be only one `.onmessage`, so that was reserved for the end user code to utilize. In the past we have had bug reports about "please be a good composable library" about this, which is why I have taken the stance of not using .onxxx on any possible publicly accessible event handlers.

.onmessage is shorter though. so I wouldn't mind it from code golfing perspective, but I'd prefer addEventListener for safety.

Comment (sbc100) [reply src/library_wasm_worker.js:142]: Out of interest if we use `.onmessage` here can our users stil used `addEventListenter()`?

I feel like we should be consistent between wasm workers and pthreads there.   I don't mind changing both but we might also need to improve the worker polyfill that we have for node

**Comment by dschuff** [comment]
This seems to be causing a failure under LSan: https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8781449586792126465/+/u/Emscripten_testsuite__LSan_/stdout
```
Aborted(Assertion failed: stackSize > tlsSize, at: /emsdk/emscripten/system/lib/wasm_worker/library_wasm_worker.c,47,emscripten_create_wasm_worker)
/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:142
   throw ex;
   ^

RuntimeError: Aborted(Assertion failed: stackSize > tlsSize, at: /emsdk/emscripten/system/lib/wasm_worker/library_wasm_worker.c,47,emscripten_create_wasm_worker)
    at abort (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:631:10)
    at ___assert_fail (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:3913:2)
    at emscripten_create_wasm_worker (<anonymous>:wasm-function[212]:0x6b28)
    at emscripten_malloc_wasm_worker (<anonymous>:wasm-function[213]:0x6b92)
    at __original_main (<anonymous>:wasm-function[46]:0xf76)
    at main (<anonymous>:wasm-function[47]:0xfb1)
    at /b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:657:20
    at callMain (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4912:13)
    at doRun (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4951:21)
    at run (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4963:3)
    at runCaller (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4901:18)
    at removeRunDependency (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:618:4)
    at /b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:889:44
    at Object.loadWasmModuleToAllWorkers (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:3753:3)
    at receiveInstance (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:889:11)
    at receiveInstantiationResult (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:897:3)
Thrown at:
    at abort (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:631:10)
    at ___assert_fail (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:3913:2)
    at emscripten_create_wasm_worker (:1:27433)
    at emscripten_malloc_wasm_worker (:1:27539)
    at __original_main (:1:3959)
    at main (:1:4018)
    at /b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:657:20
    at callMain (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4912:13)
    at doRun (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4951:21)
    at run (/b/s/w/ir/x/t/emtest_ttbbjo15/emscripten_test_lsan_0bdudi02/malloc_wasm_worker.js:4963:3)
    ```

**Comment by dschuff** [comment]
Actually I guess the problem is not new, it's just that we weren't running this test outside of the browser before, right?

**Comment by sbc100** [comment]
Right, old bug newly visible I would guess

**Pull request merged**
